### PR TITLE
Add defaults to swaplevel() parameters i and j

### LINKED ISSUE #12934: Add parameter defaults for swaplevel
I will be happy to write up a pull request, but first wanted to gauge the sanity of my suggestion:

I think that `swaplevel()` deserves default values for its parameters, just like its friends like `stack()` and `unstack()` and `sortlevel()` that also all take an initial `level` argument. I suggest:

    def swaplevel(self, i=-2, j=-1, axis=0):

This provides the greatest symmetry with the other methods that operate on levels: they all, if no level is specified, operate on the innermost level as their default.

In the very common case where there are only two levels to the multi-index anyway, this would reduce this frequent operation to simply `.swaplevel()` or `.swaplevel(axis='columns')` without, I don't think, any more loss of readability than when `stack()` or `unstack()` fail to specify the level upon which they are operating.

### PR DESCRIPTION
**Pull request by brandon-rhodes**
Closes #12934 feature request.

### CODE REVIEW

**Comment by jreback** [comment]
@brandon-rhodes I just updated master, so you will need to rebase as unrelated issue with python-dateutil will cause a test for fail :< (we are now skipping)

File: doc/source/whatsnew/v0.18.1.txt [hunk 1]
```diff
@@ -131,6 +131,8 @@ These changes conform sparse handling to return the correct types and work to ma
 
 API changes
 ~~~~~~~~~~~
+- ``.swaplevel()`` for ``Series``, ``DataFrame``, ``Panel``, and ``MultiIndex`` now features defaults for its first two parameters ``i`` and ``j`` that swap the two innermost levels of the index.
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.1.txt]: add the issue number at the end, don't add a blank line

Comment (brandon-rhodes) [inline doc/source/whatsnew/v0.18.1.txt]: To make sure I understand: are you saying that (a) even though the subsequent bullet points are separated by blank lines, the blank line beneath this bullet point should disappear? or (b) that when I go in to the code to add the issue number, I should keep it on the same line and not let it land by accident on a subsequent line?

Comment (jreback) [inline doc/source/whatsnew/v0.18.1.txt]: oh I just meant no need to add the line in between. its doesn't matter much as sphinx will ignore it, but just trying to be consistent (though I think there are some extra lines there anyhow....)

**Comment by jreback** [comment]
did u mean to close this?

**Comment by brandon-rhodes** [comment]
No. I am used to the Phabricator work flow, and this pull request is teaching me the GitHub workflow :) Stand by!

**Comment by brandon-rhodes** [comment]
@jreback Okay, I pulled and pushed some stuff, and all the code reappeared. :) I added the issue number, and rebased against your latest master — let me know how things look!

**Comment by jreback** [comment]
can u update tests for MultiIndex as well

**Comment by jreback** [comment]
pls squash as well

File: pandas/core/frame.py [hunk 2]
```diff
@@ -3384,7 +3384,7 @@ def nsmallest(self, n, columns, keep='first'):
         """
         return self._nsorted(columns, n, 'nsmallest', keep)
 
-    def swaplevel(self, i, j, axis=0):
+    def swaplevel(self, i=-2, j=-1, axis=0):
         """
         Swap levels i and j in a MultiIndex on a particular axis
```
Comment (jreback) [inline pandas/core/frame.py]: I think it's worth noting in the doc strings (of alll of these) that the defaults were added in 0.18.1

you can use a versionadded directive

**Comment by brandon-rhodes** [comment]
Oh — GitHub doesn't let you squash as you land the pull request? Yet another difference with the workflow I'm used to. Yes, I'll do the squashing myself, and also try addressing your two comments — thanks!

**Comment by jreback** [comment]
yes I will squash on merge - since we rebase its usually a bit easier to use logical commits and squash

**Comment by brandon-rhodes** [comment]
Just saw your comment — had already squashed on my end :) Per the Sphinx docs, I went with ``versionchanged`` instead, since it seemed closer to the intent of a change like this. I also found the only test I could find where the multiindex tests used swaplevel and supplemented it. Let me know what you think! (Also: wow, your integration tests take a while!)

**Comment by jreback** [comment]
@brandon-rhodes lgtm.

any comments? @sinhrks @shoyer @jorisvandenbossche @TomAugspurger

**Comment by shoyer** [comment]
This seems reasonable enough to me. :+1:

**Comment by TomAugspurger** [comment]
👍

**Comment by jreback** [comment]
thanks @brandon-rhodes

**Comment by brandon-rhodes** [comment]
Thanks for considering this contribution from me! :)

**Pull request closed without merging**